### PR TITLE
[Do not merge]Add azure managed application, get application definition API

### PR DESCRIPTION
**Pull request by dandanwang0320**
Reference code
ARM\AzureUX\ARM\src\providers\Roles\Providers.Feature\Controllers\ApplicationDefinitionArtifactController.cs

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json [hunk 1]
```diff
@@ -0,0 +1,1436 @@
...
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Solutions/applicationDefinitions/{applicationDefinitionName}/applicationArtifacts/{applicationArtifactName}": {
```
Comment (nschonni) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]:
```suggestion
    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Solutions/applicationDefinitions/{applicationDefinitionName}/applicationArtifacts/{applicationArtifactName}": {
```

**Review by raych1 — CHANGES_REQUESTED** [review]
@dandanwang0320 , pls resolve the CI failures.

**Comment by dandanwang0320** [comment]
> @dandanwang0320 , pls resolve the CI failures.

I think these errors are unrelated to this change. The new changes are in commit 75cf949b7decc905128278984369c2387c8f7a1a

Also I have another PR 5889 that works on the same folder that should fix the model validations.
So I did not make the same fixes here.

**Comment by raych1** [comment]
> @dandanwang0320 , pls resolve the CI failures.
> 
> I think these errors are unrelated to this change. The new changes are in commit 75cf949
> Also I have another PR 5889 that works on the same folder that should fix the model validations.
> So I did not make the same fixes here.

@dandanwang0320 , it doens't allow duplicated operation_id - `ApplicationDefinitions_Get` defined. operation_id has to be unique.

**Comment by raych1** [comment]
@dandanwang0320 , can you reference the [errors](https://dev.azure.com/azure-sdk/public/_build/results?buildId=80881&view=logs&jobId=8d2f9c49-cd83-5a9d-f3fe-2fd651afa742&j=f10f5372-40ed-54fc-caa4-b1cb5bd1252b) to fix README.md file to include the referenced files.

**Comment by dandanwang0320** [comment]
Should be fixed now. Also fixed model validation errors in example files. 

> @dandanwang0320 , can you reference the [errors](https://dev.azure.com/azure-sdk/public/_build/results?buildId=80881&view=logs&jobId=8d2f9c49-cd83-5a9d-f3fe-2fd651afa742&j=f10f5372-40ed-54fc-caa4-b1cb5bd1252b) to fix README.md file to include the referenced files.

**Comment by raych1** [comment]
@Tiano2017 @rajshah11 @vivsriaus, can you review this PR?

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json [hunk 2]
```diff
@@ -494,6 +488,65 @@
...
+            "maxLength": 64
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
```
Comment (ravbhatnagar) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: please include the schema for this. Isnt it strongly typed?

**Review by ravbhatnagar — COMMENTED** [review]
@dandanwang0320 - please take a look

**Comment by raych1** [comment]
@dandanwang0320 , please resolve the comments.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json (hunk 2 above)
Comment (dandanwang0320) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: No, this one returning blob content directly. See our code base in applicationdefinitionartifactcontroller.cs
The returned content will have different schema according to the input applicationArtifactName.

Comment (raych1) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2019-03-01/managedapplications.json]: @ravbhatnagar , please review the comment @dandanwang0320 replied.

**Comment by ravbhatnagar** [comment]
@dandanwang0320 - can you please confirm with Manohar whether this API needs to be documented. If not, please close the PR.

**Pull request closed without merging**